src/visitable.cpp

```diff
@@ -139,6 +139,20 @@ bool visitable<vehicle_cursor>::has_quality( const std::string &qual, int level,
     return qty <= 0 ? true : has_quality_internal( *this, qual, level, qty ) == qty;
 }
 
+template <>
+bool visitable<Character>::has_quality( const std::string &qual, int level, int qty ) const
+{
+    auto self = static_cast<const Character *>( this );
+
+    for( const auto &bio : self->my_bionics ) {
+        if( bio.get_quality( qual ) >= level ) {
+            qty--;
```

Comment:
Ok, but what about a conditional return? You already have a branch there so it's not likely to affect performance but would allow an early return?